### PR TITLE
feat: CRUD for Contribution Links in Admin Resolver

### LINKED ISSUE #1921: 🚀 [Feature] Contribution Link CRUD in Admin Resolver
## 🚀 Feature
 * Allow the basic CRUD (the D stands for soft delete!) operations for contribution links with admin rights (we want to have this in the admin resolver)
 * Write unit tests for this resolver
 * Seed at least one creation link

### PR DESCRIPTION
**Pull request by Mogge**
## 🍰 Pullrequest
add CRUD for contribution links in admin resolver

### Issues
- fixes #1921

### Todo

- [X] None

### CODE REVIEW

File: backend/src/graphql/resolver/AdminResolver.ts [hunk 1]
```diff
@@ -460,6 +466,99 @@ export class AdminResolver {
...
+    return new ContributionLink(dbContributionLink)
+  }
+
+  @Authorized([RIGHTS.LIST_CONTRIBUTION_LINKS])
+  @Query(() => ContributionLinkList)
+  async listContributionLinks(
+    @Args()
+    { currentPage = 1, pageSize = 5, order = Order.DESC }: Paginated,
+  ): Promise<ContributionLinkList> {
+    const [links, count] = await DbContributionLink.findAndCount({
```
Comment (clauspeterhuebner) [inline backend/src/graphql/resolver/AdminResolver.ts:506]: Add an info-log statement

File: backend/src/graphql/resolver/AdminResolver.ts [hunk 2]
```diff
@@ -460,6 +466,99 @@ export class AdminResolver {
...
+    })
+    return {
+      links: links.map((link: DbContributionLink) => new ContributionLink(link)),
+      count,
+    }
+  }
+
+  @Authorized([RIGHTS.DELETE_CONTRIBUTION_LINK])
+  @Mutation(() => Date, { nullable: true })
+  async deleteContributionLink(@Arg('id', () => Int) id: number): Promise<Date | null> {
```
Comment (clauspeterhuebner) [inline backend/src/graphql/resolver/AdminResolver.ts:519]: Add an info-log statement

File: backend/src/graphql/resolver/AdminResolver.ts [hunk 3]
```diff
@@ -460,6 +466,99 @@ export class AdminResolver {
...
+      name,
+      memo,
+      cycle,
+      validFrom,
+      validTo,
+      maxAmountPerMonth,
+      maxPerCycle,
+    }: ContributionLinkArgs,
+    @Arg('id', () => Int) id: number,
+  ): Promise<ContributionLink> {
```
Comment (clauspeterhuebner) [inline backend/src/graphql/resolver/AdminResolver.ts:545]: Add an info-log statement

File: backend/src/graphql/resolver/AdminResolver.ts [hunk 4]
```diff
@@ -460,6 +466,99 @@ export class AdminResolver {
...
+    const dbContributionLink = await DbContributionLink.findOne(id)
+    if (!dbContributionLink) {
+      logger.error(`Contribution Link not found to given id: ${id}`)
+      throw new Error('Contribution Link not found to given id.')
+    }
```
Comment (clauspeterhuebner) [inline backend/src/graphql/resolver/AdminResolver.ts:550]: Add business validations analog to the creation method

**Review by clauspeterhuebner — CHANGES_REQUESTED** [review]

File: backend/src/graphql/resolver/AdminResolver.test.ts [hunk 5]
```diff
@@ -1593,4 +1598,360 @@ describe('AdminResolver', () => {
...
+              )
+            })
+
+            it('updated the DB record', async () => {
+              await expect(DbContributionLink.findOne(linkId)).resolves.toEqual(
+                expect.objectContaining({
+                  id: linkId,
+                  name: 'Dokumenta 2023',
+                  memo: 'Danke für deine Teilnahme an der Dokumenta 2023',
+                  // amount: '400',
```
[the remark this answers is not in the document]
Comment (Tirokk) [reply backend/src/graphql/resolver/AdminResolver.test.ts:1901]: same same

File: backend/src/seeds/graphql/mutations.ts [hunk 6]
```diff
@@ -137,3 +137,84 @@ export const deletePendingCreation = gql`
...
+      name
+      memo
+      code
+      link
+      createdAt
+      validFrom
+      validTo
+      maxAmountPerMonth
+      cycle
+      maxPerCycle
```
Comment (Tirokk) [inline backend/src/seeds/graphql/mutations.ts:172]: Why the `id` is not given back here?
The only thing we can not figure out otherwise …

**Review by Tirokk — DISMISSED** [review]
Even a huge feature!
Really good and big work! @Mogge 🍀

But I have some questions and would like changes …

File: backend/src/graphql/resolver/AdminResolver.ts [hunk 7]
```diff
@@ -460,6 +466,99 @@ export class AdminResolver {
...
+      amount,
+      name,
+      memo,
+      cycle,
+      validFrom,
+      validTo,
+      maxAmountPerMonth,
+      maxPerCycle,
+    }: ContributionLinkArgs,
+  ): Promise<ContributionLink> {
```
[the remark this answers is not in the document]
Comment (Mogge) [reply backend/src/graphql/resolver/AdminResolver.ts:484]: Time is running fast this week. We need this PR to get merged.

File: backend/src/graphql/resolver/AdminResolver.ts (hunk 1 above)
Comment (Mogge) [reply backend/src/graphql/resolver/AdminResolver.ts:506]: Can we do this in a later stage?

File: backend/src/seeds/graphql/mutations.ts (hunk 6 above)
Comment (Mogge) [reply backend/src/seeds/graphql/mutations.ts:172]: Right!

**Review by clauspeterhuebner — DISMISSED** [review]

**Review by clauspeterhuebner — APPROVED** [review]

**Pull request merged**